lib/_http_outgoing.js

```diff
@@ -119,6 +121,25 @@ Object.defineProperty(OutgoingMessage.prototype, 'writableFinished', {
   }
 });
 
+Object.defineProperty(OutgoingMessage.prototype, 'writableObjectMode', {
+  get: function() {
+    return false;
+  }
+});
+
+Object.defineProperty(OutgoingMessage.prototype, 'writableLength', {
+  get: function() {
+    return this.outputSize + (this.socket ? this.socket.writableLength : 0);
+  }
+});
+
+Object.defineProperty(OutgoingMessage.prototype, 'writableHighWaterMark', {
+  get: function() {
+    assert(!this.socket || this.socket.writableHighWaterMark === HIGH_WATER_MARK);
```

Comment:
What if the high water mark for the socket *does* differ from the default?

Reply:
@addaleax: It never does... at least not as far as I've been able to find.

Reply:
@ronag It definitely does if the default value is changed for some reason. If you want a reproduction:

```diff
diff --git a/test/common/duplexpair.js b/test/common/duplexpair.js
index 0783aeb86110..bc968f85c39a 100644
--- a/test/common/duplexpair.js
+++ b/test/common/duplexpair.js
@@ -8,7 +8,7 @@ const kOtherSide = Symbol('Other');
 
 class DuplexSocket extends Duplex {
   constructor() {
-    super();
+    super({ highWaterMark: 100 });
     this[kCallback] = null;
     this[kOtherSide] = null;
   }
diff --git a/test/parallel/test-http-generic-streams.js b/test/parallel/test-http-generic-streams.js
index bc28d1fcd463..c05a50e9fc76 100644
--- a/test/parallel/test-http-generic-streams.js
+++ b/test/parallel/test-http-generic-streams.js
@@ -19,6 +19,7 @@ const MakeDuplexPair = require('../common/duplexpair');
   const req = http.request({
     createConnection: common.mustCall(() => clientSide)
   }, common.mustCall((res) => {
+    console.log('HWM =', res.socket.writableHighWaterMark)
     res.setEncoding('utf8');
     res.on('data', common.mustCall((data) => {
       assert.strictEqual(data, testData);
```

```sh
$ ./node test/parallel/test-http-generic-streams.js 
HWM = 100
```

Reply:
@addaleax updated the code

Reply:
While no socket is assigned the default buffering is used. See, https://github.com/nodejs/node/pull/29019.